flink-runtime/src/main/java/org/apache/flink/runtime/security/factories/SecurityFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.security.factories;
+
+/**
+ * Base class for all security related security factory instantiable.
+ */
+public interface SecurityFactory {
```

Comment:
Why do we need a base class here? If it's only for the loader code, we can add a generic parameter to that method. I don't see any code that is actually polymorphic, where we would pass in either a `SecurityContextFactory` or `SecurityModuleFactory`, i.e. where we would need a common base class.

Reply:
yes. that's correct I was trying to make the loader code of `private static SecurityFactory findFactoryInternal` simpler. but yes I think it makes sense to use a generic type param.